ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt

```diff
@@ -214,36 +207,34 @@ constructor(
    * appropriate zoom level and subsequent ones only pan the map.
    */
   @OptIn(ExperimentalCoroutinesApi::class)
-  private suspend fun updateCameraPositionOnLocationChange() {
+  private fun updateCameraPositionOnLocationChange(): Flow<CameraUpdateRequest> =
     locationLock
       .flatMapLatest { enabled ->
         getLocationUpdates()
           .map { it.toCoordinates() }
           .filter { enabled.getOrDefault(false) }
           .withIndex()
       }
-      .collect { (index, coordinates) ->
+      .map { (index, coordinates) ->
```

Comment:
Can we extract this lambda into its own function? eg `onLocationUpdate()`?

Reply:
Done